tools/run_tests/xds_k8s_test_driver/framework/infrastructure/gcp/api.py

```diff
@@ -260,29 +261,68 @@ class OperationError(Error):
     https://cloud.google.com/apis/design/design_patterns#long_running_operations
     https://github.com/googleapis/googleapis/blob/master/google/longrunning/operations.proto
     """
+    api_name: str
+    name: str
+    metadata: str
+    code_name: code_pb2.Code
+    error: status_pb2.Status
 
-    def __init__(self, api_name, operation_response, message=None):
+    def __init__(self, api_name: str, response: dict):
         self.api_name = api_name
-        operation = json_format.ParseDict(
-            operation_response,
-            Operation(),
-            ignore_unknown_fields=True,
-            descriptor_pool=error_details_pb2.DESCRIPTOR.pool)
+
+        # Operation.metadata field is Any specific to the API. It may not be
+        # present in the default descriptor pool, and that's expected.
+        # To avoid json_format.ParseError, handle it separately.
+        self.metadata = response.pop('metadata', {})
+
+        # Must be after removing metadata field.
+        operation: Operation = self._parse_operation_response(response)
         self.name = operation.name or 'unknown'
         self.code_name = code_pb2.Code.Name(operation.error.code)
         self.error = operation.error
-        # Collect error details packed as Any without parsing concrete types.
-        self.error_details = [
-            text_format.MessageToString(any_error, as_one_line=True)
-            for any_error in self.error.details
-        ]
-        if message is None:
-            message = (f'{api_name} operation "{self.name}" failed. Error '
-                       f'code: {self.error.code} ({self.code_name}), '
-                       f'message: {self.error.message}, '
-                       f'details: {self.error_details}')
-        self.message = message
-        super().__init__(message)
+        super().__init__()
+
+    @staticmethod
+    def _parse_operation_response(operation_response: dict) -> Operation:
+        try:
+            return json_format.ParseDict(
+                operation_response,
+                Operation(),
+                ignore_unknown_fields=True,
+                descriptor_pool=error_details_pb2.DESCRIPTOR.pool)
+        except (json_format.Error, TypeError) as e:
+            # Swallow parsing errors if any. Building correct OperationError()
+            # is more important than losing debug information. Details still
+            # can be extracted from the warning.
+            logger.warning(
```

Comment:
nit: Maybe INFO is enough? WARNING log nudges people to make changes to do the right thing. In this case, we can't do much about it (well, we certainly don't want to import every single google api protos).

Reply:
With the change I made (not unpacking metadata `Any`), I don't expect this code branch to occur normally. In fact, if it does happen, it needs to be investigated, and, potentially, fixed.

Reply:
Cool. Then let's use WARNING.